GoRogue/MapGeneration/Region.cs

```diff
@@ -72,62 +118,21 @@ public Region(Lines.Algorithm algorithm, IComponentCollection? components, param
         /// <param name="corners">The corners of the region</param>
         public Region(params Point[] corners) : this(corners, Lines.Algorithm.DDA) { }
 
-        private Region(List<Point> corners, Lines.Algorithm algorithm, IComponentCollection? components)
+        /// <summary>
+        /// Returns a new Region using the provided area
+        /// </summary>
+        /// <param name="area">This region's Area</param>
+        /// <param name="components">A component collection for this region</param>
+        public Region(PolygonArea area, IComponentCollection? components = null)
```

Comment:
This constructor generates a bunch of nullability warnings because it thinks that `_area` is not assigned a non-null value.

There is some annotation that could be used on the `Area` property to fix this, but my thought is actually to just bypass the property entirely, and set `_area = area` instead.  There's no reason to attempt to fire the event in the constructor since there couldn't possibly be any handlers for it anyway.